homeassistant/components/homekit/type_security_systems.py

```diff
@@ -166,5 +167,18 @@ def async_update_state(self, new_state: State) -> None:
                 hass_state,
                 current_state,
             )
-        if (target_state := HASS_TO_HOMEKIT_TARGET.get(hass_state)) is not None:
+
+        if hass_state == STATE_ALARM_ARMING:
+            if (hass_next_state := new_state.attributes.get("next_state")) is not None:
```

Comment:
`next_state` is only defined by `manual.alarm_control_panel`. We can only use attributes that are defined by the base entity class here as it won't work with any other integration

https://developers.home-assistant.io/docs/core/entity/alarm-control-panel/